data/migrations/20190423144343_roles.js

```diff
@@ -0,0 +1,28 @@
+
+exports.up = function(knex, Promise) {
+    return knex.schema
+        .createTable('roles', tbl => {
+            tbl.increments('role_id');
+
+            tbl
+            .string('name', 128)
+            .notNullable();
+
+            tbl
+            .text('description');
+
+            tbl
+            .integer('project_id')
+            .unsigned()
+            .notNullable()
+            .references('_id')
```

Comment:
I haven't tested it myself, but did you intend for the roles to reference the projects?
I thought the students were going to reference the roles then projects reference the students.